samples/simulated-annealing/Training.ipynb

```diff
@@ -0,0 +1,193 @@
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Declare parameters and model hyperparameters\n",
+    "Components may declare (and use) these default parameters:\n",
+    "- dataset\n",
+    "- target\n",
+    "\n",
+    "Use these parameters to load/save datasets, models, metrics, and figures with the help of [PlatIAgro SDK](https://platiagro.github.io/sdk/). <br />\n",
+    "You may also declare custom parameters to set when running an experiment.\n",
+    "\n",
+    "Select the hyperparameters and their respective values to be used when using the component:\n",
+    "- date\n",
+    "- group\n",
+    "- alpha"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {
+    "tags": [
+     "parameters"
+    ]
+   },
+   "outputs": [],
+   "source": [
+    "# parameters\n",
+    "dataset = \"eucalyptus-2.csv\" #@param {type:\"string\"}\n",
+    "target = \"Utility\" #@param {type:\"string\"}\n",
+    "date = None #@param {type:\"string\"}\n",
+    "group = [\"Sp\"] #@param {type:\"list\"}\n",
+    "\n",
+    "# Increase solution search\n",
+    "alpha = 1 #@param {type:\"integer\"}"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Load dataset\n",
+    "\n",
+    "Import and put the whole dataset in a pandas.DataFrame."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "from platiagro import load_dataset\n",
+    "\n",
+    "df = load_dataset(name=dataset)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Load metadata about the dataset\n",
+    "For example, below we get the feature type for each column in the dataset. (eg. categorical, numerical, or datetime)"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "import numpy as np\n",
+    "import pandas as pd\n",
+    "from platiagro import stat_dataset\n",
+    "\n",
+    "metadata = stat_dataset(name=dataset)\n",
+    "featuretypes = metadata[\"featuretypes\"]\n",
+    "df_features = pd.DataFrame(featuretypes)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Call Auto Featuring"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "from simulated import SimulatedAnnealing\n",
+    "\n",
+    "simulated = SimulatedAnnealing(df,\n",
+    "                                df_features,\n",
+    "                                target,\n",
+    "                                group,\n",
+    "                                date,\n",
+    "                                alpha)\n",
+    "\n",
+    "#Pre-process\n",
+    "simulated.preprocess()\n",
+    "\n",
+    "# Autofeat\n",
+    "df, features, transformations = simulated.auto_feat()"
```

Comment:
@lucaslzl Já aconteceu isso aqui com você? :O
![boom](https://user-images.githubusercontent.com/1791381/83922382-99591680-a756-11ea-80ae-e6e6e93ddeed.PNG)

Reply:
Nunca tinha visto esse erro.  Mas, pelo que li pode ser algum uma função aplicada em um valor, ou que gerou um resultado, fora do intervalo esperado. Portanto, adicionei um tratamento de erro.

https://stackoverflow.com/questions/45462743/typeerror-when-using-df-apply-pandas

Reply:
Adicionei um tratamento de erro.

Reply:
Ainda aconteceu comigo. Pelo que vi tem vários valores '?' em colunas numéricas do eucalyptus e o algoritmo está fazendo coisas assim: `sin('?')` `square('?')`

Acho legal levar esse ponto pra Graziella e perguntar se devemos manter sem este tratamento.

Reply:
Eu não sei porque mas os dados vazios estão sendo alterados pra "?". Se eu leio direto pelo pandas o arquivo retorna os NaN. Ou seja, no processo ele foi alterado para "?".

Reply:
Eu adicionei o código a seguir para forçar a coluna vazia até a gente descobrir o motivo da mudança.
for column in df.columns:
    df[df[column] == "?"] = ""